api/python/provisioner/commands/set_swupgrade_repo.py

```diff
@@ -333,6 +376,54 @@ def dynamic_validation(self, params: inputs.SWUpgradeRepo, targets: str):  # noq
 
         return repo.metadata
 
+    @staticmethod
+    def _setup_python_index(repo: inputs.SWUpgradeRepo):
+        """
+        Setup Python index
+
+        Parameters
+        ----------
+        repo: inputs.SWUpgradeRepo
+            SW upgrade repository parameters
+
+        Returns
+        -------
+
+        """
+        def setup_default():
+            _config = ConfigParser()
+
+            _config['global'] = {
```

Comment:
During initial setup we apply [this SLS](https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/api/python/provisioner/srv/salt/repos/pip_config.sls). And you may see that it's close to what you have here but different actually. So your logic will override that.

Let's have the following assumption:
- custom pip.conf is already on place

So you need only to append extra index urls.

Reply:
Yes, I made this change based on logic in the salt state file provided by your link. I had the assumption that I have no guarantees that pip.conf will be configuration before `set_swupgrade_repo` command execution. 
So based on the assumption provided by you, I will delete this logic.

Also, I had thoughts to use salt states to manage `pip.conf` ini configuration and put that logic to `repos/_macros.sls`:
https://docs.saltproject.io/en/latest/ref/states/all/salt.states.ini_manage.html
it turned out to be inconvenient because we need to handle cases when `extra-index-url` section has already existed in `pip.conf`